ale_linters/c/clang.vim

```diff
@@ -8,22 +8,33 @@ function! ale_linters#c#clang#GetExecutable(buffer) abort
     return ale#Var(a:buffer, 'c_clang_executable')
 endfunction
 
-function! ale_linters#c#clang#GetCommand(buffer) abort
-    let l:paths = ale#c#FindLocalHeaderPaths(a:buffer)
+function! ale_linters#c#clang#GetCommand(buffer, output) abort
+    let l:cflags = []
+    if !empty(a:output)
+        let l:cflags = join(ale#c#ParseMakefile(a:buffer, join(a:output, '\n')), ' ')
```

Comment:
I think you meant to use `ParseCFlags` here.

These lines for parsing the flags are all the same, so you could write one function in the `autoload/ale/c.vim` which does both and just write `let l:cflags = ale#c#GetFlags(a:buffer, a:output)` here.